src/fetch-wrapper.ts

```diff
@@ -40,7 +40,7 @@ export default function fetchWrapper(
     method: requestOptions.method,
     body: requestOptions.body,
     headers: requestOptions.headers as HeadersInit,
-    signal: (requestOptions as any).signal,
+    signal: requestOptions.request?.signal,
```

Comment:
Should we have a test to cover this?

Reply:
There used to be a test that would have tested for this exact scenario, but we removed it as we didn't feel it was necessary.
See the discussion that led to its removal https://github.com/octokit/request.js/pull/599#discussion_r1247225312

Reply:
Ready to merge then 🚀